src/core/shape_inference/include/utils.hpp

```diff
@@ -17,6 +17,7 @@ void copy_shape_infer(const OpType* op, const std::vector<T>& input_shapes, std:
 template <class OpType, class T>
 void first_input_passthrough_infer(const OpType* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
     NODE_VALIDATION_CHECK(op, output_shapes.size() == 1, "Incorrect number of output shapes");
+    NODE_VALIDATION_CHECK(op, input_shapes.size() >= 1, "Incorrect number of input shapes");
```

Comment:
I didn't expect that. But there is a whole lot of sense to check for number of outputs too. We only set output shape for the first one.
```suggestion
    NODE_VALIDATION_CHECK(op, output_shapes.size() == 1 && input_shapes.size() >= 1, "Incorrect number of input and output shapes");
```

Reply:
OK, changed